pkg/api/norman/customization/gke/listers.go

```diff
@@ -195,3 +195,36 @@ func listZones(ctx context.Context, cap *Capabilities) ([]byte, int, error) {
 
 	return data, http.StatusOK, err
 }
+
+func listSharedNetworks(ctx context.Context, cap *Capabilities) ([]byte, int, error) {
+	if cap.ProjectID == "" {
+		return nil, httperror.ServerError.Status, fmt.Errorf("projectId is required")
+	}
```

Comment:
This might be moved to helper, will need to check that sort of thing when we rebase.

Reply:
Oh wait, these should be handled now before this point

Reply:
Pretty sure this needs to be handled here, same as the others https://github.com/rancher/rancher/pull/31829/files#diff-18b89e110c42b6e252c7b898f1c84f058011c923d7af5f01125f4085acba8550R87

Reply:
It should be handled inside getCloudCredential, before this point: https://github.com/rancher/rancher/blob/master/pkg/api/norman/customization/gke/handler.go#L144

Reply:
Oops, fixed